GeneralsMD/Code/GameEngine/Source/Common/RTS/Money.cpp

```diff
@@ -100,6 +102,46 @@ void Money::deposit(UnsignedInt amountToDeposit, Bool playSound)
 	}
 }
 
+// ------------------------------------------------------------------------------------------------
+void Money::setStartingCash(UnsignedInt amount)
+{
+	m_startingCash = amount;
+	m_money = amount;
+	m_currentBucket = 0;
+	m_lastBucketFrame = 0;
+	for (UnsignedInt i = 0; i < 60; ++i)
+		m_incomeBuckets[i] = 0;
+}
+
+// ------------------------------------------------------------------------------------------------
+void Money::updateIncomeBucket()
+{
+	UnsignedInt frame = TheGameLogic->getFrame();
+	UnsignedInt lastSec = m_lastBucketFrame / LOGICFRAMES_PER_SECOND;
+	UnsignedInt curSec = frame / LOGICFRAMES_PER_SECOND;
+	UnsignedInt diff = (curSec > lastSec) ? curSec - lastSec : 0;
+	if (diff > 0)
+	{
+		if (diff > 60)
+			diff = 60;
+		for (UnsignedInt i = 0; i < diff; ++i)
+		{
+			m_currentBucket = (m_currentBucket + 1) % 60;
```

Comment:
I think this can be moved out the loop with `m_currentBucket = (m_currentBucket + diff) % 60;`